fuse-products/src/main/java/software/tnb/product/cq/application/LocalQuarkusApp.java

```diff
@@ -18,18 +21,27 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
-import java.util.stream.Collectors;
+import java.util.Optional;
 
 public class LocalQuarkusApp extends QuarkusApp {
     private static final Logger LOG = LoggerFactory.getLogger(LocalQuarkusApp.class);
     private Process appProcess;
-    private final AbstractIntegrationBuilder<?> integrationBuilder;
 
     public LocalQuarkusApp(AbstractIntegrationBuilder<?> integrationBuilder) {
         super(integrationBuilder);
 
-        this.integrationBuilder = integrationBuilder;
         endpoint = new Endpoint(() -> "http://localhost:" + integrationBuilder.getPort());
+
+        Optional<Customizer> restCustomizer = integrationBuilder.getCustomizers().stream().filter(c -> c instanceof RestCustomizer).findFirst();
+        // For local quarkus app, the HTTP request will fail when the endpoint is not ready, so check if an exception was raised or not
+        restCustomizer.ifPresent(customizer -> readinessCheck = () -> {
```

Comment:
`customizer -> readinessCheck = () -> {` this is looking like Java no more :D 
How about 
```java
        restCustomizer.ifPresent(customizer -> {
            this.readinessCheck = () -> {
                try {
                    HTTPUtils.getInstance().get(getEndpoint() + ((RestCustomizer) customizer).getHealthCheckPath());
                    return true;
                } catch (Exception e) {
                    return false;
                }
            };
        });
```

Reply:
/shrug it it what idea created